sound/soc/sof/intel/hda.c

```diff
@@ -45,55 +45,49 @@
  * Helper function to set up a DAI widget in the DSP and then send the updated DAI config during
  * hw_params or send the updated DAI_CONFIG and then free the DAI widget during hw_free
  */
-int hda_ctrl_dai_widget_setup(struct snd_sof_dev *sdev, struct snd_sof_dai *sof_dai,
-			      struct sof_ipc_dai_config *config, bool setup, int dir)
+int hda_ctrl_dai_widget_setup(struct snd_soc_dapm_widget *w, bool setup)
 {
-	struct snd_sof_widget *swidget;
+	struct snd_sof_widget *swidget = w->dobj.private;
+	struct snd_soc_component *component = swidget->scomp;
+	struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(component);
```

Comment:
reorder the definition shape and move "struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(component);" to first one